lib/util.ex

```diff
@@ -67,27 +77,33 @@ defmodule Util do
       Authorization: "Bearer " <> auth
     ]
 
-    {:ok, resp} = HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers)
-
-    resp.body
-    |> JSON.decode!()
-    |> Map.get("data")
-    |> Map.get("tournaments")
-    |> Map.get("nodes")
-    # Finesse data into our own non-start.gg format.
-    |> Enum.map(fn sgg_data ->
-      %{
-        "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
-        "name" => sgg_data["name"],
-        "location" => %{
-          "lat" => sgg_data["lat"],
-          "lng" => sgg_data["lng"]
-        },
-        "start_time" => sgg_data["startAt"],
-        "end_time" => sgg_data["endAt"],
-        "updated_at" => sgg_data["updatedAt"],
-        "url" => "https://start.gg/" <> (sgg_data["slug"])
-      }
-    end)
+    case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
+      {:ok, resp} ->
+        try do
+          resp.body
+          |> JSON.decode!
+          |> Map.get("data")
+          |> Map.get("tournaments")
+          |> Map.get("nodes")
+          # Finesse data into our own non-start.gg format.
+          |> Enum.map(fn sgg_data ->
+            %{
+              "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
+              "name" => sgg_data["name"],
+              "location" => %{
+                "lat" => sgg_data["lat"],
+                "lng" => sgg_data["lng"]
+              },
+              "start_time" => sgg_data["startAt"],
+              "end_time" => sgg_data["endAt"],
+              "updated_at" => sgg_data["updatedAt"],
+              "url" => "https://start.gg/" <> (sgg_data["slug"])
+            }
+            end )
+          rescue
+            _e -> raise "Ran into an issue while parsing the response body in get_tourney_page function."
```

Comment:
The most common error that comes up here is a BadMapError that comes from the Map.get calls, more than likely the first or second Map.get call that tries to pull from `nil` instead of a valid map value.